File: clients/admin/src/features/departments/types/employee-level.ts
export interface EmployeeLevelDto {
  id: string;
  companyId: string;
  name: string;
  code: string;
  rank: number;
  description?: string | null;
}

export interface CreateEmployeeLevelRequest {
  companyId: string;
  name: string;
  code: string;
  rank: number;
  description?: string;
}

File: src/Organization.Api/Domain/Employees/EmployeeLevel.cs
using StarterKit.Shared.Entities;

namespace StarterKit.Organization.Api.Domain.Employees;

public class EmployeeLevel : AuditableEntity
{
    public string CompanyId { get; set; } = null!;

    public string Name { get; set; } = null!;

    public string Code { get; set; } = null!;

    public int Rank { get; set; }

    public string? Description { get; set; }
}
